src/hooks/posters.ts

```diff
@@ -45,27 +48,32 @@ export const usePosters = (posterHallId: string) => {
 
   const [liveFilter, setLiveFilter] = useState<boolean>(false);
 
-  const hasFilters = searchQuery || liveFilter;
-
-  const searchResultPosterVenues = useMemo(() => {
-    const searchedPosterVenues = (() => {
-      if (!searchQuery) return posterVenues;
-
-      return new Fuse(posterVenues, {
+  const fuseVenues = useMemo(
+    () =>
+      new Fuse(posterVenues, {
         keys: ["poster.title", "poster.authorName", "poster.categories"],
-      })
-        .search(searchQuery)
-        .map((fuseSeachItem) => fuseSeachItem.item);
-    })();
-
-    if (liveFilter)
-      return searchedPosterVenues.filter((posterVenue) => posterVenue.isLive);
-
-    return searchedPosterVenues;
-  }, [posterVenues, searchQuery, liveFilter]);
+      }),
+    [posterVenues]
+  );
+
+  const searchedPosterVenues = useMemo(() => {
+    if (!searchQuery) return posterVenues;
+
+    return fuseVenues
+      .search(searchQuery)
+      .map((fuseSearchItem) => fuseSearchItem.item);
+  }, [searchQuery, fuseVenues, posterVenues]);
```

Comment:
[NB] I know that `posterVenues` is here so that we can return it when `!searchQuery`, but it will also mean that this memo will break whenever `fuseVenues` or `posterVenues` changes. Since `fuseVenues` will change when `posterVenues` changes anyway, this might not be a big deal (depending on how React batches changes), but I wonder if we can get the same result by just using `fuseVenues`?

I was hoping there might be a method that returns the underlying collection directly, but it seems maybe not:

- https://fusejs.io/api/methods.html

We could hold it in a `ref` to avoid re-rendering, but that feels like overkill/premature optimisation at this stage.

So.. probably fine to leave it as is.

Reply:
I guess even if it does break memo, it may end up returning the same underlying collection anyway, which would have the same reference; so it wouldn't necessarily 'cascade' to breaking memo on downstream dependents.